### ISSUE TITLE
Pickling issue with Sire.Units.GeneralUnit

### ISSUE THREAD

**Issue by lohedges** [michellab/Sire#388]
As reported by @xiki-tempula:

```python
import pickle
import BioSimSpace.Sandpit.Exscientia as BSS
a = BSS.Units.Energy.kcal_per_mol / BSS.Units.Area.nanometer2
pickle.dump(a, open('unit.p', 'wb'))
b = pickle.load(open('unit.p', 'rb'))
b / a
```

This should result in a dimensionless value, i.e. a float of 1.0. Bizarrely we get the following:

```python
a == b
True

a.dimensions() == b.dimensions()
True
```

However, when checking the internal Sire unit:

```python
a._to_sire_unit()
0.01 M T-2 Q-1

b._to_sire_unit()
0.01
```

The unpickled GeneralUnit's internal Sire unit is dimensionless, even though the dimensions are reported correctly!

**Comment by chryswoods** [comment]
This was caused by a missing streaming operator. I've added this in `feat_web` in [this commit](https://github.com/michellab/Sire/commit/24b64ae1eff9b1943481f73ce57cceed00e9e752).

**Comment by lohedges** [comment]
Closed in 858d72fc58b311160065c8613d72395be4a70946.
